discovery-provider/alembic/trigger_sql/handle_repost.sql

```diff
@@ -1,146 +1,298 @@
+	    and track_id = new.repost_item_id;
+	end if;
+	else milestone_name := 'PLAYLIST_REPOST_COUNT';
+	update aggregate_playlist
+	set repost_count = (
+	        SELECT count(*)
+	        FROM reposts r
+	        WHERE
+	            r.is_current IS TRUE
+	            AND r.is_delete IS FALSE
+	            AND r.repost_type = new.repost_type
+	            AND r.repost_item_id = new.repost_item_id
+	    )
+	where
+	    playlist_id = new.repost_item_id returning repost_count into new_val;
+	if new.is_delete IS FALSE then
+	select
+	    playlist_owner_id into owner_user_id
+	from playlists
+	where
+	    is_current
+	    and playlist_id = new.repost_item_id;
+	end if;
+	end if;
+	-- create a milestone if applicable
+	select
+	    new_val into milestone
+	where
+	    new_val in (
+	        10,
+	        25,
+	        50,
+	        100,
+	        250,
+	        500,
+	        1000,
+	        5000,
+	        10000,
+	        20000,
+	        50000,
+	        100000,
+	        1000000
+	    );
+	if new.is_delete = false
+	and milestone is not null
+	and owner_user_id is not null then
+	insert into
+	    milestones (
+	        id,
+	        name,
+	        threshold,
+	        blocknumber,
+	        slot,
+	        timestamp
+	    )
+	values (
+	        new.repost_item_id,
+	        milestone_name,
+	        milestone,
+	        new.blocknumber,
+	        new.slot,
+	        new.created_at
+	    ) on conflict
+	do nothing;
+	insert into
+	    notification (
+	        user_ids,
+	        type,
+	        specifier,
+	        group_id,
+	        blocknumber,
+	        timestamp,
+	        data
+	    )
+	values (
+	        ARRAY [owner_user_id],
+	        'milestone',
+	        owner_user_id,
+	        'milestone:' || milestone_name || ':id:' || new.repost_item_id || ':threshold:' || milestone,
+	        new.blocknumber,
+	        new.created_at,
+	        json_build_object(
+	            'type',
+	            milestone_name,
+	            'threshold',
+	            milestone
+	        )
+	    ) on conflict
+	do nothing;
+	end if;
+	-- notify followees of the reposter who have reposted the same content
+	-- within the last month
+	begin if new.is_delete is false
+	and new.is_repost_repost is true then
+	with
+	    followee_repost_repost_ids as (
+	        select user_id
+	        from reposts r
+	        where
+	            r.repost_item_id = new.repost_item_id
+	            and new.created_at + INTERVAL '-1 month' < r.created_at
+	            and new.created_at > r.created_at
+	            and r.user_id in (
+	                select
+	                    followee_user_id
+	                from follows
+	                where
+	                    follower_user_id = new.user_id
+	            )
+	    )
+	insert into
+	    notification (
+	        blocknumber,
+	        user_ids,
+	        timestamp,
+	        type,
+	        specifier,
+	        group_id,
+	        data
+	    )
+	values (
+	        new.blocknumber,
+	        ARRAY(
+	            SELECT user_id
```

Comment:
what happens if this is empty?

Reply:
i left a note about this in the description. it just inserts a row with an empty array, which seems harmless since there will be no one to notify but also is kinda ugly data so can see why this isn't ideal behavior. i couldn't figure out how to not insert if empty though so if you have suggestions pls lmk